test/e2e/nbthread_test.go

```diff
@@ -0,0 +1,128 @@
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const nbthreadRouterDeployTimeout time.Duration = 1 * time.Minute
+
+// NOTE: This test will mutate the default ingresscontroller and the
+// "cluster" ingress configuration.
+func TestRouteNbthreadIngressController(t *testing.T) {
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, defaultName, defaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	ic, err := nbthreadGetIngressController(t, kclient, 1*time.Minute)
+	if err != nil {
+		t.Fatalf("failed to get ingress controller: %v", err)
+	}
+
+	// by default, the router deployment should have ROUTER_THREADS set to 4
+	if err := nbthreadWaitForRouterDeployment(t, kclient, nbthreadRouterDeployTimeout, ic, 4); err != nil {
+		t.Fatalf("expected router deployment to have spec.threading.count unset: %v", err)
+	}
+
+	// set spec.threading.count to a nonstandard value, 7
+	if err := nbthreadSetThreadCount(t, kclient, 1*time.Minute, 7, ic); err != nil {
```

Comment:
Perhaps we should generally define `4` as a constant in the deployment.go then do `2 * _DEFAULT_` just in case we ever change the "default" to 7 - we (or future "we") wouldn't notice any regression in that case.

Reply:
done, thanks for the suggestion